### DISCUSSION TITLE
Does SkyWalking support the Javalin Rest API framework?

### DISCUSSION THREAD

**Discussion by whitecamel-ctrl** [apache/skywalking#8331] [Q&A] [answered]
Does SkyWalking support the Javalin Rest API framework?

We found that the trace of the service using Javalin framework was incomplete after it was connected to SkyWalking.

like this:
![image](https://user-images.githubusercontent.com/39118151/147207718-db6e7013-da6c-4cb8-a006-c13a08e8cf30.png)

**Comment by wu-sheng** [comment, marked as answer]
Supported list doc is provided in the Java agent. Check there.

**Reply by whitecamel-ctrl** [reply]
Yes, I have checked the doc and found no support for Javalin. Does Skywalking plan to support this framework in the future?

**Reply by wu-sheng** [reply]
The community doesn't plan to support anything. If someone wants to contribute, then it is there. Plugins are out of committers' scope. It is countless and changing every day.

**Reply by whitecamel-ctrl** [reply]
ok thx!
